docs/fsharp/style-guide/conventions.md

```diff
@@ -641,3 +641,19 @@ module Networking =
 ```
 
 In summary, the pitfall with Type Abbreviations is that they are **not** abstractions over the types they are abbreviating. In the previous example, `BufferSize` is just an `int` under the covers, with no additional data, nor any benefits from the type system besides what `int` already has.
+
+An alternative approach to using type abbreviations to represent a domain is to use single-case discriminated unions. The previous sample can be modeled as follows:
+
+```fsharp
+type BufferSize = BufferSize of int
+```
+
+If you write code that operates in terms of `BufferSize` and its underlying value, you need to construct one rather than pass in any arbitrary integer:
+
+```fsharp
+module Networking =
+    ...
+    let send data (BufferSize size) =
+    ...
+
+This reduces the likelihood that an arbitrary integer is mistakenly passed into the `send` function, because the caller must construct a `BufferSize` type to wrap a value before calling the function.
```

Comment:
```suggestion
This reduces the likelihood of mistakenly passing an arbitrary integer into the `send` function, because the caller must construct a `BufferSize` type to wrap a value before calling the function.
```